_routes.php

```diff
@@ -1373,80 +1427,73 @@ function ($request, $response) {
+                    ->withHeader('Location', $this->router->pathFor('objectslend_enable_objects'));
+            } elseif (isset($post['print_list'])) {
                 return $response
                     ->withStatus(301)
                     ->withHeader('Location', $this->router->pathFor('objectslend_objects_print'));
+            } elseif (isset($post['print_objects'])) {
+                return $response
+                    ->withStatus(301)
+                    ->withHeader('Location', $this->router->pathFor('objectslend_objects_printobject'));
+            } else {
+                $this->flash->addMessage(
+                    'error_detected',
+                    _T("No action was found. Please contact plugin developpers.")
+                );
             }
-
-            $this->flash->addMessage(
-                'error_detected',
-                _T("No action was found. Please contact plugin developpers.")
-            );
         } else {
             $this->flash->addMessage(
                 'error_detected',
                 _T("No object was selected, please check at least one.")
             );
         }
-
         return $response
             ->withStatus(301)
             ->withHeader('Location', $this->router->pathFor('objectslend_objects'));
     }
 )->setName('objectslend_batch-objectslist')->add($authenticate);
 
-$this->get(
-    __('/objects', 'objectslend_routes') . __('/remove', 'routes'),
-    function ($request, $response) {
-        $filters =  $this->session->objectslend_filter_objects;
 
-        $data = [
-            'id'            => $filters->selected,
-            'redirect_uri'  => $this->router->pathFor('objectslend_objects')
-        ];
-
-        // display page
-        $this->view->render(
-            $response,
-            'confirm_removal.tpl',
-            array(
-                'type'          => _T("Object"),
-                'mode'          => $request->isXhr() ? 'ajax' : '',
-                'page_title'    => _T('Remove objects'),
-                'message'       => str_replace(
-                    '%count',
-                    count($data['id']),
-                    _T('You are about to remove %count objects.')
-                ),
-                'form_url'      => $this->router->pathFor('objectslend_doremove_object'),
-                'cancel_uri'    => $this->router->pathFor('objectslend_objects'),
-                'data'          => $data
-            )
-        );
-        return $response;
-    }
-)->setName('objectslend_remove_objects')->add($authenticate);
 
 $this->get(
     __('/objects', 'objectslend_routes') . __('/print', 'objectslend_routes') . '[/{id:\d+}]',
     function ($request, $response, $args) {
         $lendsprefs = new Preferences($this->zdb);
 
-        if (isset($this->session->objectslend_filter_objects)) {
-            $filters =  $this->session->objectslend_filter_objects;
-        } else {
+        if (isset($args['id'])) {
```

Comment:
Cette modification est incorrecte. La condition `isset($this->session->objectslend_filter_objects)` devrait rester

Reply:
c'est corrigé